devimint/src/tests.rs

```diff
@@ -2096,10 +2097,25 @@ pub async fn cannot_replay_tx_test(dev_fed: DevFed) -> Result<()> {
         CLIENT_START_AMOUNT - CLIENT_SPEND_AMOUNT
     );
 
-    cmd!(double_spend_client, "reissue", double_spend_notes)
-        .run()
-        .await
-        .expect_err("double spend must fail");
+    // TODO(support:v0.2): remove
+    if VersionReq::parse(">=0.3.0-alpha")?.matches(&fedimint_cli_version) {
+        let reissue_error = cmd!(double_spend_client, "reissue", double_spend_notes)
+            .expect_err_json()
+            .await?
+            .get("error")
+            .expect("json error contains error field")
+            .as_str()
+            .context("not a string")?
+            .to_owned();
+
+        assert!(reissue_error.contains("The transaction had an invalid input"));
```

Comment:
Feels like this could be an `assert_error` function, much like `expect_err_json`, that takes a predicate closure. A specialization of it may just take a string that should be contained in the error message (the only thing we can check for now).

Reply:
Good call 👍 

Opened a followup PR https://github.com/fedimint/fedimint/pull/4605